gm_dotnet_managed/GmodNET/GameConsoleWriter.cs

```diff
@@ -13,9 +14,53 @@ internal class GameConsoleWriter : TextWriter
         public override string NewLine => "\n";
         public override Encoding Encoding => Encoding.UTF8;
 
-#pragma warning disable CA2101
+        public GameConsoleWriter(ILua lua)
+        {
+            if (OperatingSystem.IsWindows())
+            {
+                Msg = Msg_Tier0;
+            }
+            else if (OperatingSystem.IsLinux())
+            {
+                lua.PushSpecial(SPECIAL_TABLES.SPECIAL_GLOB);
+                lua.GetField(-1, "game");
+                lua.GetField(-1, "IsDedicated");
+                lua.MCall(0, 1);
+                bool is_dedicated = lua.GetBool(-1);
+                lua.Pop(3);
+
+                if (is_dedicated)
+                {
+                    Msg = Msg_Tier0;
+                }
+                else
+                {
+                    Msg = Msg_Tier0_Client;
+                }
+            }
+            else if (OperatingSystem.IsMacOS())
+            {
+                Msg = Msg_Tier0_MacOS;
+            }
+            else Msg = (string msg) => { throw new PlatformNotSupportedException(); };
+        }
+
+#pragma warning disable CA2101 // workaround for https://github.com/dotnet/roslyn-analyzers/issues/5009
+
         [DllImport("tier0", CallingConvention = CallingConvention.Cdecl)]
-        private static extern void Msg([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
+        private static extern void Msg_Tier0([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
+
+        [DllImport("tier0_client", CallingConvention = CallingConvention.Cdecl)]
+        private static extern void Msg_Tier0_Client([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
+
+        [DllImport("GarrysMod_Signed/Contents/MacOS/libtier0.dylib", CallingConvention = CallingConvention.Cdecl)]
+        private static extern void Msg_Tier0_MacOS([MarshalAs(UnmanagedType.LPUTF8Str)] string msg);
```

Comment:
We can't use DllImport on macOS due to the game app bundle structure. Here is an example of the fix which worked for me: https://gist.github.com/GlebChili/a6d14a7fc7f8970bd49a6b4573b5eacc

Reply:
whoa!
`delegate*` usage!

```cs
                    Msg = (message) =>
                    {
                        byte[] string_array = Encoding.UTF8.GetBytes(message);
                        fixed (byte* ptr = &string_array[0])
                        {
                            msg_func(ptr);
                        }
                    };
```
i probably can find a better solution for that.